soc/arm/st_stm32/stm32f0/soc.c

```diff
@@ -36,7 +39,9 @@
  */
 void relocate_vector_table(void)
 {
-#ifndef CONFIG_IS_BOOTLOADER
+#if defined(CONFIG_SW_VECTOR_RELAY) || defined(CONFIG_SW_VECTOR_RELAY_CLIENT)
```

Comment:
Please also rework the above function's brief to explicit the assumptions and intentions.
I agree all symbols are documented but the addition makes it rather complex.

Reply:
Done